### ISSUE TITLE
NCCL error when pmapping across 2 GPUs

### ISSUE THREAD

**Issue by jimmysmith1919** [google/jax#15628]
### Description

I get a 'xla.gpu.all_reduce' failed when attempting to use lax.pmean and jax.pmap across 2 GPUs. The code works fine on a single GPU. Basic example:

```
import jax
import jax.numpy as np

x = np.arange(2)
y = jax.pmap(lambda x: jax.lax.pmean(x, 'i'), axis_name='i')(x)

2023-04-16 15:44:18.280050: E external/xla/xla/pjrt/pjrt_stream_executor_client.cc:2432] Execution of replica 1 failed: INTERNAL: Failed to execute XLA Runtime executable: run time error: custom call 'xla.gpu.all_reduce' failed: external/xla/xla/service/gpu/nccl_all_reduce_thunk.cc:244: NCCL operation ncclGroupEnd() failed: unhandled cuda error.
2023-04-16 15:44:18.280275: E external/xla/xla/pjrt/pjrt_stream_executor_client.cc:2432] Execution of replica 0 failed: INTERNAL: Failed to execute XLA Runtime executable: run time error: custom call 'xla.gpu.all_reduce' failed: external/xla/xla/service/gpu/nccl_all_reduce_thunk.cc:244: NCCL operation ncclGroupEnd() failed: unhandled cuda error.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/groups/swl1/jsmith14/miniconda3/envs/icl_env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/home/groups/swl1/jsmith14/miniconda3/envs/icl_env/lib/python3.9/site-packages/jax/_src/api.py", line 1861, in cache_miss
    out = map_bind_continuation(execute(*tracers))
  File "/home/groups/swl1/jsmith14/miniconda3/envs/icl_env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/home/groups/swl1/jsmith14/miniconda3/envs/icl_env/lib/python3.9/site-packages/jax/_src/interpreters/pxla.py", line 1916, in __call__
    results = self.xla_executable.execute_sharded(input_bufs)
jax._src.traceback_util.UnfilteredStackTrace: jaxlib.xla_extension.XlaRuntimeError: INTERNAL: Failed to execute XLA Runtime executable: run time error: custom call 'xla.gpu.all_reduce' failed: external/xla/xla/service/gpu/nccl_all_reduce_thunk.cc:244: NCCL operation ncclGroupEnd() failed: unhandled cuda error.: while running replica 0 and partition 0 of a replicated computation (other replicas may have failed as well).

The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.

--------------------

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
jaxlib.xla_extension.XlaRuntimeError: INTERNAL: Failed to execute XLA Runtime executable: run time error: custom call 'xla.gpu.all_reduce' failed: external/xla/xla/service/gpu/nccl_all_reduce_thunk.cc:244: NCCL operation ncclGroupEnd() failed: unhandled cuda error.: while running replica 0 and partition 0 of a replicated computation (other replicas may have failed as well).
```

I have attempted to try different combinations and versions of CUDA, CUDNN, and NCCL with no luck.

```
nvcc --version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Mon_Sep_13_19:13:29_PDT_2021
Cuda compilation tools, release 11.5, V11.5.50
Build cuda_11.5.r11.5/compiler.30411180_0
```

### What jax/jaxlib version are you using?

0.4.8/0.4.7

### Which accelerator(s) are you using?

NVIDIA A100 GPU

### Additional system info

python 3.9.16, Linux, cuda/11.5.0, cudnn/8.6.0.163, nccl/2.11.4

### NVIDIA GPU info

+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 530.30.02              Driver Version: 530.30.02    CUDA Version: 12.1     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                  Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf            Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-80GB           On | 00000000:81:00.0 Off |                    0 |
| N/A   29C    P0               80W / 500W|      0MiB / 81920MiB |      0%   E. Process |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA A100-SXM4-80GB           On | 00000000:C1:00.0 Off |                    0 |
| N/A   28C    P0               81W / 500W|      0MiB / 81920MiB |      0%   E. Process |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+

**Comment by mjsML** [comment]
@jimmysmith1919 did you try to upgrade to CUDA 11.8 or later? note that with [0.4.8 CUDA 11.4 would be already deprecated](https://github.com/google/jax#pip-installation-gpu-cuda-installed-locally-harder) , if you did try with the later versions of the stack, are you facing the exact same error / logs?

**Comment by jimmysmith1919** [comment]
Hi @mjsML, just to clarify I am using CUDA 11.5 not 11.4 (it is unclear from the JAX documentation if the deprecated message applies to 11.5 or not also). These issues are occurring in an HPC cluster environment. CUDA 11.8 is not available as a module, however CUDA 11.7.1 and CUDA 12.0.0 are available. But if I try loading either of these, when I load cuDNN/8.6.0.163 (the latest version of cuDNN available on the cluster), it automatically downgrades my CUDA installation to 11.5.0. Thus I was attempting to use CUDA/11.5.0, cuDNN/8.6.0.163, and NCCL/2.11.4 with JAX 0.4.8. Is there anything obviously wrong with this combo? As I mentioned, my pmapped functions work on a single GPU, but I get the NCCL issue when trying to pmap across multiple gpus.  Thoughts?
